src/createPersistor.js

```diff
@@ -52,17 +51,47 @@ export default function createPersistor (store, config) {
           return
         }
 
-        let key = storesToProcess[0]
-        let storageKey = createStorageKey(key)
-        let endState = transforms.reduce((subState, transformer) => transformer.in(subState, key), stateGetter(store.getState(), key))
-        if (typeof endState !== 'undefined') storage.setItem(storageKey, serializer(endState), warnIfSetError(key))
+        persistCurrentStateForKey(state, storesToProcess[0])
         storesToProcess.shift()
       }, debounce)
     }
 
     lastState = state
   })
 
+  function findStoresToProcess (state) {
+    let keys = []
+    stateIterator(state, (subState, key) => {
+      if (!passWhitelistBlacklist(key)) return
+      let newState = stateGetter(state, key)
+      if (lastWritten[key] === newState) return
+      if (stateGetter(lastState, key) === newState) return
+      keys.push(key)
+    })
+
+    return keys
+  }
+
+  function persistCurrentStateForKey (state, key) {
+    let storageKey = createStorageKey(key)
+    let currentState = stateGetter(state, key)
+    lastWritten[key] = currentState
+    let endState = transforms.reduce((subState, transformer) => transformer.in(subState, key), currentState)
+    if (typeof endState === 'undefined') return null
+    return storage.setItem(storageKey, serializer(endState), warnIfSetError(key))
+  }
+
+  function persistCurrentState () {
```

Comment:
would be more performant to just write the existing keys that are pending (everything else is gauranteed to be up to date)

i.e.
```js
keys.forEach((key) => {
  persistCurrentStateForKey(state, key)
})
keys = []
```